docs/Optimizations.md

```diff
@@ -67,6 +67,12 @@ Below you can see the list of currently supported graph optimizations:
     This optimization performs a classic CSE with a goal of avoiding of any
     results that were computed already.
 
+  * Optimization of ReduceMean nodes
+
+    This optimization performs substitions of ReduceMean with AvgPool node if
+    the reduce parameters are suitable: input is 4D with last two dimensions
+    to be reduced, and keepdims parameter is true.  
```

Comment:
Is it actually necessary for keepDims to be true? You can always perform the optimization and just add a reshape. It doesn't look like this is reflected in the optimization code itself either.

Reply:
Idea was indeed to do the optimization only if keepDims is true. That was ReduceMean we had problems with and were trying to change only what's required. But also, that was making code simpler - no keepDims passed to this node and no need to reshape and remove dimensions (since AvgPool keeps dimensions).
I can add `keepDims `to `BatchedReduceMeanNode` specification, and add Reshape that erases two dimensions, into this optimization.